scripts/check_docs.py

```
@@ -0,0 +1,126 @@
+                continue
+
+            if not destination.exists():
+                errors.append(f"{relative}:{line}: missing link target: {display}")
+                continue
+
+            if (
+                fragment
+                and destination.is_file()
+                and destination.suffix.lower() == ".md"
+            ):
+                if fragment not in anchors(destination):
```

Ruff (0.16.0)

[warning] Use a single `if` statement instead of nested `if` statements

Combine `if` statements using `and`

(SIM102)